### PR TITLE
Feature/moses/create notification on message recieved

### PR DESCRIPTION
**Pull request by SapeleD3**

### CODE REVIEW

File: functions/controllers/message/readMessages.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+const getMessages = async (req, res, db) => {
+    const chats = [];
+    try {
+        const messages = await db
+            .collection('messages')
+            .where('roomId', '==', req.params.roomId)
+            .get();
+        messages.docs.forEach(async mes => {
+            chats.push(mes.data());
+        });
```
Comment (EngrEric) [inline functions/controllers/message/readMessages.js:41]: Why are you using `async` here?
This is more neat and optimal
```
const chats = messages.docs.map(mes =>mes.data());
```

File: functions/controllers/message/sendMessages.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+const successMessage = {
+    success1: res => {
+        res.status(OK).json({
+            message: messageSent,
+            status: success,
+        });
+    },
+};
```
Comment (EngrEric) [inline functions/controllers/message/sendMessages.js:24]: I see that you have repeated this several places.
Try to have a global error message and success message functions and accept custom message as the case maybe.
```
const errors = {
    error2: (res, errorType, message) => {
        res.status(errorType).json({
            message,
            status: error,
        });
    },
};

const successMessage = {
    success1: (res, message) => {
        res.status(OK).json({
            message,
            status: success,
        });
    },
};
```
this way you can reuse it anywhere in your code

File: functions/controllers/message/sendMessages.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+const sendMessages = async (req, res, db) => {
+    try {
+        const room = await db
+            .doc(`/rooms/${req.params.recieverId + req.user.uid}`)
+            .get();
+        if (room.data() === undefined) {
+            return createRoom(req, res, db);
+        }
```
Comment (EngrEric) [inline functions/controllers/message/sendMessages.js:66]: why do you create room when the user requested to send message?
if the room does not exist, its better to say hey `this room is not found` then from the frontend you can recommend creating the room.

**Review by EngrEric — CHANGES_REQUESTED** [review]
Require minor changes

File: functions/controllers/message/readMessages.js (hunk 1 above)
Comment (SapeleD3) [reply functions/controllers/message/readMessages.js:41]: I have to use for each so I can work on the array.. And I did not want the flow to skip that before execution

File: functions/controllers/message/sendMessages.js (hunk 3 above)
Comment (SapeleD3) [reply functions/controllers/message/sendMessages.js:66]: Every chat session creates a chat session .. that's the room.. Its like start a convoy between this people if they've not spoken prior to that moment

Comment (EngrEric) [reply functions/controllers/message/sendMessages.js:66]: got you

File: functions/controllers/message/readMessages.js (hunk 1 above)
Comment (EngrEric) [reply functions/controllers/message/readMessages.js:41]: But you are only pushing docs data to the array chats.

**Pull request merged**